setup.py

```diff
@@ -8,9 +8,10 @@
         "pytest-xdist",  # multi-process runner
         "pytest-cov",  # Coverage analyzer plugin
         "hypothesis>=6.2.0,<7.0",  # Strategy-based fuzzer
+        "pytest-mock",  # Test mocker
```

Comment:
This is not technically needed right now

Reply:
this will be needed when we make the testing though before this PR can be resolved?

Reply:
i think we need this so that I can add the functional test we talked about previously last week